helper/resource/testing_sets.go

```diff
@@ -14,25 +14,46 @@ const (
 	sentinelIndex = "*"
 )
 
-// TestCheckTypeSetElemNestedAttrs is a TestCheckFunc that accepts a resource
-// name, an attribute path, which should use the sentinel value '*' for indexing
-// into a TypeSet. The function verifies that an element matches the whole value
-// map.
+// TestCheckTypeSetElemNestedAttrs ensures a subset map of values is stored in
+// state for the given name and key combination of attributes nested under a
+// list or set block. Use this TestCheckFunc in preference over non-set
+// variants to simplify testing code and ensure compatibility with indicies,
+// which can easily change with schema changes. State value checking is only
+// recommended for testing Computed attributes and attribute defaults.
 //
-// You may check for unset keys, however this will also match keys set to empty
-// string. Please provide a map with at least 1 non-empty value.
+// For managed resources, the name parameter is combination of the resource
+// type, a period (.), and the name label. The name for the below example
+// configuration would be "myprovider_thing.example".
 //
-//   map[string]string{
-//	     "key1": "value",
-//       "key2": "",
-//   }
+//     resource "myprovider_thing" "example" { ... }
 //
-// Use this function over SDK provided TestCheckFunctions when validating a
-// TypeSet where its elements are a nested object with their own attrs/values.
+// For data sources, the name parameter is a combination of the keyword "data",
+// a period (.), the data source type, a period (.), and the name label. The
+// name for the below example configuration would be
+// "data.myprovider_thing.example".
+//
+//     data "myprovider_thing" "example" { ... }
+//
+// The key parameter is an attribute path in Terraform CLI 0.11 and earlier
+// "flatmap" syntax. Keys start with the attribute name of a top-level
+// attribute. Use the sentinel value '*' to replace the element indexing into
+// a list or set. The sentinel value can be used for each list or set index, if
+// there are multiple lists or sets in the attribute path.
+//
+// The values parameter is the map of attribute names to attribute values
+// expected to be nested under the list or set.
+//
+// You may check for unset nested attributes, however this will also match keys
+// set to empty string. Use a map with at least 1 non-empty value.
```

Comment:
`to empty string` -> `to an empty string`

Reply:
```suggestion
// set to an empty string. Use a map with at least 1 non-empty value.
```